docs/naps/0-nap-process.md

```diff
@@ -0,0 +1,297 @@
+
+### Review and Resolution
+
+NAPs are discussed in Zulip, on image.sc, and on GitHub. The possible paths
+of the status of NAPs are as follows:
+
+```{image} _static/nap-flowchart.png
+```
+
+All NAPs should be created with the `Draft` status.
+
+The author of the NAP should periodically update the NAP with new points
+both against and in favor of the NAP raised in discussion.
+
+Eventually, after discussion, there may be a consensus that the NAP
+should be accepted – see the next section for details. At this point
+the status becomes `Accepted`.
+
+Once a NAP has been `Accepted`, the reference implementation must be
+completed. When the reference implementation is complete and incorporated
+into the main source code repository, the status will be changed to
+`Final`.
+
+To allow gathering of additional design and interface feedback before
+committing to long term stability for a language feature or standard
+library API, a NAP may also be marked as "Provisional". This is short for
+"Provisionally Accepted", and indicates that the proposal has been accepted
+for inclusion in the reference implementation, but additional user feedback
+is needed before the full design can be considered "Final". Unlike regular
+accepted NAPs, provisionally accepted NAPs may still be Rejected or
+Withdrawn even after the related changes have been included in a release.
+
+Wherever possible, it is considered preferable to reduce the scope of a
+proposal to avoid the need to rely on the "Provisional" status (e.g. by
+deferring some features to later NAPs), as this status can lead to version
+compatibility challenges in the wider ecosystem.
+
+A NAP can also be assigned status `Deferred`. The NAP author or a core
+developer can assign the NAP this status when no progress is being made on
+the NAP.
+
+A NAP can also be `Rejected`. Perhaps after all is said and done it
+was not a good idea. It is still important to have a record of this
+fact. The `Withdrawn` status is similar---it means that the NAP author
+themselves has decided that the NAP is actually a bad idea, or has
+accepted that a competing proposal is a better alternative.
+
+When a NAP is `Accepted`, `Rejected`, or `Withdrawn`, the NAP should be
+updated accordingly. In addition to updating the status field, at the very
+least the `Resolution` header should be added with a link to the relevant
+post on the discussion forum.
+
+NAPs can also be `Superseded` by a different NAP, rendering the
+original obsolete. The `Replaced-By` and `Replaces` headers
+should be added to the original and new NAPs respectively.
+
+Process NAPs may also have a status of `Active` if they are never
+meant to be completed, e.g. NAP 0 (this NAP).
+
+### How a NAP becomes Accepted
+
+A NAP is `Accepted` by consensus of all interested contributors. We need a
+concrete way to tell whether consensus has been reached. When you think a
+NAP is ready to accept, start a topic on the Zulip naps channel with a
+subject like:
```

Comment:
slight preference for this to be on github (or somewhere other than zulip).  zulip history is very hard to search, and appears to be ephemeral?

Reply:
+1 to this I feel like GitHub is a better place to put this discussion for the reasons Talley mentioned, and also because it *feels* more public

Reply:
yeah +1 GitHub not zulip being the final record. Fine to start convo in zulip as maybe easier to build consensus with realtime convo, but place of record should be GitHub, possibly with links to any essential Zulip convos. We could use either GitHub issue or GitHub discussion, I'd be open to either

Reply:
I'll say three things. First, the final record of discussion is the NAP itself. Second, Zulip is as far as I know *not* ephemeral — old Zulip links resolve fine for me. Search on Zulip is not great but it's not the access mechanism here: the thread will be permalinked from the NAP. (And it's important to note that loginless Zulip is currently being tested in self-hosted instances and coming in the hosted version, zulip/zulip#4817.) Third, and I think this is the main reason that other projects *don't* use GitHub, Zulip is an open source platform, which means if they close up shop we can grab our data and self-host and everything will still work. If GitHub decides to shut down their API we are left in the lurch.

Now, it's absolutely valid to say, screw it, In GitHub We Trust, and just go with it, as we implicitly do with virtually all our development. Clearly if GH closes up we have bigger problems than the complete NAP discussion record! 😂 But I just wanted to write down the rationale for why none of the other SKIP/NEP/PEP mechanisms use GH as the main record of these decisions and have us take the decision with that rationale in mind.